Feature/arpa 53#3

MichaelKaiser1 opened the pull request:
Translation (en,de) not working. Help requested

MichaelKaiser1 committed:
added captcha to login component

MichaelKaiser1 committed:
Revert "added captcha to login component"

This reverts commit e9a767476663297b2621ed2766d038b20f6f6353.

MichaelKaiser1 committed:
Revert "Revert "added captcha to login component""

This reverts commit dbc1ae6d13115e2fc867270799ee29358b325fe4.

MichaelKaiser1 committed:
minor change

MichaelKaiser1 committed:
solved lint

VILLAN3LL3 commented on src/environments/environment.prod.ts:
```diff
@@ -1,6 +1,9 @@
   production: true,
-    api: {
+  captcha: {
+    key: '6Le-g3QaAAAAAJmIN-s37441yuZPBp1M7nqUwtED'
```
Sind das geheime Schlüssel? Dann sollten wir die auf keinen Fall einchecken

MichaelKaiser1 replied:
Ist der öffentliche Schlüssel

VILLAN3LL3 commented on src/app/components/onboarding/login/login.component.html:
```diff
@@ -15,14 +15,15 @@
             <label for="float-input">{{ 'PASSWORD' | translate }}</label>
             <i id="eyeButton" class="p-ml-2 pi {{hide ? 'pi-eye-slash' : 'pi-eye'}}" (click)="hide = !hide"></i>
           </p>
+          <p-captcha #cpt siteKey = "{{captchaKey}}"  theme = 'dark' language="{{this.language}}" (onResponse)="showResponse($event)" initCallback="loadCaptcha"></p-captcha>
```
Verwende die Attribute lieber so: `[siteKey]="captchaKey" [language]="language"`

VILLAN3LL3 commented on src/app/components/onboarding/login/login.component.ts:
```diff
@@ -19,12 +21,18 @@ export class LoginComponent implements OnInit, OnDestroy {
+  captchaKey = environment.captcha.key;
+  script: any;
 
   constructor(formBuilder: FormBuilder,
```
Das Captcha sollte glaub ich ins Registrierungsformular und nicht ins Login :)

MichaelKaiser1 replied:
Hatte mit Michael besprochen dass ich es hier rein mach. Geht aber natürlich auch auf der anderen Seite

wolfgangroese replied:
brauchen wir das im Login @MichaelSpielmann ? Hätte es jetzt auch eher auf der Registerseite verortet um Spam-Registrierungen zu vermeiden.. beim login is ja wurscht denk ich, oder?

VILLAN3LL3 commented on src/app/app.module.ts:
```diff
@@ -103,6 +104,7 @@ export function tokenGetter(): string | null {
       progressBar: true,
       positionClass: 'toast-top-full-width'
     }),
+    CaptchaModule,
```
Kannst du das ins Primeng Module aufnehmen anstatt ins app module?

MichaelKaiser1 committed:
import captcha in prime-ng.module

MichaelKaiser1 committed:
implemented different option

Transalation and Validation works but browser console throws exception:
Unhandled Promise rejection: reCAPTCHA has already been rendered in this element ; Zone: <root> ; Task: Promise.then ; Value: Error: reCAPTCHA has already been rendered in this element

MichaelKaiser1 committed:
minor changes

MichaelKaiser1 commented:
Habe jetzt noch eine andere Variante implementiert. Übersetzung funktioniert. Allerdings wird in der Konsole im Browser ein Error geworfen
test läuft leider nicht durch :(
https://github.com/Enngage/ngx-captcha#readme
https://www.npmjs.com/package/ngx-captcha

MichaelKaiser1 committed:
Merge branch 'main' into feature/ARPA-53#3

MichaelKaiser1 commented:
Könnte sein dass das Problem von der Google API kommt
https://github.com/DethAriel/ng-recaptcha/issues/123
